src/lib/types/types.ts

```diff
@@ -60,6 +60,7 @@ export type ThrelteContext = {
   size: Readable<Size>
   pointer: Writable<Vector2>
   pointerOverCanvas: Writable<boolean>
+  pointerInvalidated: boolean
```

Comment:
I think this property should go either in the `ThrelteRenderContext` or a new, separate `ThrelteInteractivityContext` where we would extract these properties from the `ThrelteRootContext`:
```ts
addRaycastableObject: (obj: Object3D) => void
removeRaycastableObject: (obj: Object3D) => void
addInteractiveObject: (obj: Object3D) => void
removeInteractiveObject: (obj: Object3D) => void
interactiveObjects: Set<Object3D>
raycastableObjects: Set<Object3D>
raycaster: Raycaster
lastIntersection: Intersection<Object3D<Event>> | null
```
but that's probably something for later.

Reply:
Moved it to renderCtx for now. Not sure what you mean with extract from root context.